### PR TITLE
fix isAppAlreadyInstalledo error handling

### PR DESCRIPTION
**Pull request by patrykw-splunk**
### Description

This PR aims to fix  FIPS Error Handling: 
Modified `isAppAlreadyInstalled()` and `installApp()` functions to properly handle FIPS provider informational messages in stderr by treating them as non-error conditions rather than installation failures. 
Also added fanout CR status synchronization (`installPhaseManager()`) to ensure that when app installation fails after max retries on any replica, the main PhaseInfo is properly updated to reflect the overall deployment failure state.

#### FIPS error handling

- Modified `isAppAlreadyInstalled()` function to detect and ignore FIPS provider informational messages in stderr instead of treating them as errors
- Updated `installApp()` function to handle FIPS messages during app installation, preventing false installation failures
- Added FIPS message detection logic that checks for "FIPS provider enabled" string and logs as informational rather than error

#### Fanout CR Status Management

- **Enhanced failure handling for fanout CRs** (Standalone with multiple replicas) to properly update main PhaseInfo when max retries are reached
- **Added synchronization logic** to ensure overall deployment status reflects individual replica failures
- **Improved status reporting** by setting DeployStatus to Error when legitimate failures occur after retry exhaustion

### Testing and Verification

TestRunLocalScopedPlaybook

✅ Fixed Test3: Now uses proper exit codes (fmt.Errorf("exit status 2")) instead of stderr-only failures
✅ Fixed Test4: Properly simulates grep exit code 1 for app-not-enabled scenarios
✅ Fixed Test6: Updated to test stderr content transparency

TestPremiumAppScopedPlaybook:
✅ Fixed Test3: Uses real error conditions with proper exit codes
✅ Fixed Test4: Corrected install failure simulation with err != nil
✅ Fixed Test5: Updated to test generic stderr content handling
✅ Fixed Test7: App-already-installed scenario with stderr content 

TestIsAppAlreadyInstalled:
✅ Updated test cases to be FIPS-agnostic
✅ "stderr content" in test names and descriptions
✅ Tests now focus on behavior: stderr content should be ignored for error detection
✅ Comprehensive coverage: success, not-found, real errors, edge cases

### Related Issues

N/A

### PR Checklist

- [x] Code changes adhere to the project's coding standards.
- [x] Relevant unit and integration tests are included.
- [ ] Documentation has been updated accordingly.
- [x] All tests pass locally.
- [x] The PR description follows the project's guidelines.

### CODE REVIEW

File: pkg/splunk/enterprise/afwscheduler.go [hunk 1]
```diff
@@ -785,28 +793,55 @@ func isAppAlreadyInstalled(ctx context.Context, cr splcommon.MetaObject, podExec
 
 	scopedLog.Info("check app's installation state")
 
-	command := fmt.Sprintf("/opt/splunk/bin/splunk list app %s -auth admin:`cat /mnt/splunk-secrets/password`| grep ENABLED; echo -n $?", appTopFolder)
+	command := fmt.Sprintf("/opt/splunk/bin/splunk list app %s -auth admin:`cat /mnt/splunk-secrets/password`| grep ENABLED", appTopFolder)
```
Comment (kubabuczak) [inline pkg/splunk/enterprise/afwscheduler.go]: Looks like we are grepping porcelain output. Maybe there is a way of getting machine parsable output format?
